servers/vm/shm.h

```diff
@@ -0,0 +1,19 @@
+#ifndef __SHM_H__
+#define __SHM_H__
+
+#include <types.h>
+struct shm_t {
+    bool inuse;
+    int shm_id;
+    paddr_t paddr;
+    size_t len;
+};
+
+#define NUM_SHARED_MEMS_MAX 32
+static struct shm_t shared_mems[NUM_SHARED_MEMS_MAX];
```

Comment:
Nit: please avoid declaring `static` variables in a header file. I guess this line should be in `shm.c`.